app/controllers/mentorships_controller.rb

```diff
@@ -36,9 +36,26 @@ def destroy
     end
   end
 
+  def mark_completed
+    mentorship = mark_completed_policy_scope(Mentorship).find(params[:id])
+    user = current_user
+
+    if mentorship.mentor_id == user.id
+      mentorship.mentor_completed = true
+    elsif mentorship.mentee_id == user.id
+      mentorship.mentee_completed = true
+    end
+    mentorship.save
+    redirect_to :back
+  end  
+
   private
 
-  def policy_scope scope
+  def destroy_policy_scope scope
     scope.where(mentee: current_user)
   end
+
+  def mark_completed_policy_scope scope
```

Comment:
I think this approach is really good. We may be reaching the point that we are ready for something like pundit. How do you feel about implementing it? https://github.com/elabs/pundit